pkg/controller/activegate/activegate_controller.go

```diff
@@ -142,12 +142,25 @@ func (r *ReconcileActiveGate) Reconcile(request reconcile.Request) (reconcile.Re
 	}
 
 	actualStatefulSet := &appsv1.StatefulSet{}
-	reconcileResult, err := r.manageStatefulSet(desiredStatefulSet, actualStatefulSet, reqLogger)
+	reconcileResult, err := r.manageStatefulSet(reqLogger, instance, desiredStatefulSet, actualStatefulSet)
 	if reconcileResult != nil {
-		if err != nil {
-			return *reconcileResult, err
+		return *reconcileResult, err
+	}
+
+	pods, err := r.findPods(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	err = r.setVersionLabel(pods)
+	if err != nil {
+		if _, isStatusError := err.(*errors.StatusError); isStatusError {
```

Comment:
For reference there is `errors.As()` on the Go library. Although probably better to do this check before creating the StateFul set?